pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -404,6 +403,7 @@ func (c *Controller) reconcileShoot(ctx context.Context, logger *logrus.Entry, g
 		alreadyReconciledDuringThisTimeWindow      = gutil.LastReconciliationDuringThisTimeWindow(shoot)
 		regularReconciliationIsDue                 = isUpToDate && isNowInEffectiveShootMaintenanceTimeWindow && !alreadyReconciledDuringThisTimeWindow
 		reconcileAllowed                           = !failedOrIgnored && ((!reconcileInMaintenanceOnly && !confineSpecUpdateRollout(shoot.Spec.Maintenance)) || !isUpToDate || (isNowInEffectiveShootMaintenanceTimeWindow && !alreadyReconciledDuringThisTimeWindow))
+		err                                        error
```

Comment:
Is this needed?

Reply:
it' because of [this line](https://github.com/gardener/gardener/blob/38dce592e5b2aeafda39e3ccdc928a9579190ca6/pkg/gardenlet/controller/shoot/shoot_control.go#L494):
```
o.Shoot.Info, err = c.updateShootStatusOperationSuccess(ctx, gardenClient.GardenCore(), o, o.Shoot.Info, o.Shoot.SeedNamespace, &o.Seed.Info.Name, operationType)
```
and changing
```
shoot, err := c.updateShootStatusOperationStart(ctx, gardenClient.GardenCore(), shoot, shootNamespace, operationType)
...
```
to
```
if err := c.updateShootStatusOperationStart(ctx, gardenClient.Client(), shoot, shootNamespace, operationType); err != nil {
...
```